src/fftpack.f90

```diff
@@ -268,6 +270,46 @@ pure module function idct_rk(x, n, type) result(result)
         end function idct_rk
     end interface idct
 
+    !> Version: experimental
+    !>
+    !> Initialize DCT type-1
+    !> ([Specification](../page/specs/fftpack.html#dct_t1i))
+    interface dct_t1i
+        module procedure :: dcosti
```

Comment:
My goal here is to use subroutine `dcosti` through its existing interface, without having to redefine it. How can I achieve that?

Reply:
It seems to be my problem. I gave a bad example in [#34](https://github.com/fortran-lang/fftpack/issues/34#issuecomment-1449431649).

Try changing `module procedure :: dcosti` here to `procedure :: dcosti`.

`dcosti` is not an internal routine in `module`, and `module procedure` is required for internal routine overloading, but `dcosti` is an **external** routine with Fortran77 style, so `procedure` is used instead of `module procedure`. I hope I didn't say anything wrong.

Reply:
No worries @zoziha, thank you for your feedback. That makes sense; I made the changes and the build is successful.